2004-11-28  James A. Morrison  <phython@gcc.gnu.org>

	* ada/acats/run_all.sh (target_run): Run test through run_test.exp.
	* ada/acats/run_test.exp: Expect script for ACATS tests.

Index: a/gcc/testsuite/ada/acats/run_all.sh
===================================================================
--- a/gcc/testsuite/ada/acats/run_all.sh	(revision 146010)
+++ b/gcc/testsuite/ada/acats/run_all.sh	(working copy)
@@ -13,7 +13,7 @@ gccflags="-O2"
 gnatflags="-gnatws"
 
 target_run () {
-$*
+	$testdir/run_test.exp $1 > $2
 }
 
 # End of customization section.
@@ -99,7 +99,7 @@ cp $testdir/support/*.ada $testdir/suppo
 
 # Find out the size in bit of an address on the target
 target_gnatmake $testdir/support/impbit.adb >> $dir/acats.log 2>&1
-target_run $dir/support/impbit > $dir/support/impbit.out 2>&1
+target_run $dir/support/impbit $dir/support/impbit.out
 target_bit=`cat $dir/support/impbit.out`
 echo target_bit="$target_bit" >> $dir/acats.log
 
@@ -276,7 +276,7 @@ for chapter in $chapters; do
       if [ ! -x $dir/tests/$chapter/$i/$binmain ]; then
          sync
       fi
-      target_run $dir/tests/$chapter/$i/$binmain > $dir/tests/$chapter/$i/${i}.log 2>&1
+      target_run $dir/tests/$chapter/$i/$binmain $dir/tests/$chapter/$i/${i}.log
       cd $dir/tests/$chapter/$i
       cat ${i}.log >> $dir/acats.log
       egrep -e '(==== |\+\+\+\+ |\!\!\!\! )' ${i}.log > /dev/null 2>&1
Index: a/gcc/testsuite/ada/acats/run_test.exp
===================================================================
--- a/gcc/testsuite/ada/acats/run_test.exp	(revision 0)
+++ b/gcc/testsuite/ada/acats/run_test.exp	(revision 0)
@@ -0,0 +1,10 @@
+#!/usr/bin/expect -f
+
+set timeout 300
+
+spawn $argv
+expect timeout {
+       send_user "Program timed out.\n"
+       exit 1
+}
+
